syntax = "proto3";

// This defines the protocol buffers used by XBOS with WaveMQ. The schema for
// a wavemq xbos proto message is:
//   xbosproto/RootMessageType
//
// Version 1.0
package xbospb;


// Hamilton Project definitions
import "hamilton.proto";

// XBOS IoT device definitions
import "iot.proto";

// Dent meters
import "dentmeter.proto";

// System monitors
import "system_monitor.proto";

// parker refrigerator
import "parker.proto";

// c37 + other energise-related
import "c37.proto";

// Root union type
// We reserve 50 fields per imported file
message XBOS {

  //Hamilton Fields
  HamiltonData HamiltonData = 50;
  HamiltonBRLinkStats HamiltonBRLinkStats = 51;
  HamiltonBRMessage HamiltonBRMessage = 52;

  //IoT Fields
  XBOSIoTDeviceState XBOSIoTDeviceState = 100;
  XBOSIoTDeviceActuation XBOSIoTDeviceActuation = 101;
  XBOSIoTContext XBOSIoTContext = 102;

  //Vendor-specific Device Fields
  DentMeterState DentMeterState = 150;
  ParkerState parker_state = 151;

  //System Status Fields
  BasicServerStatus BasicServerStatus = 200;

  //External Protocols
  C37DataFrame C37DataFrame = 250;
}
